sdk/core/core-http/lib/policies/proxyPolicy.ts

```diff
@@ -13,22 +13,93 @@ import { WebResource } from "../webResource";
 import { Constants } from "../util/constants";
 import { URLBuilder } from "../url";
 
+function environmentProxyValue(name: string): string | undefined {
+  if (process.env[name]) {
+    return process.env[name];
+  } else if (process.env[name.toLowerCase()]) {
+    return process.env[name.toLowerCase()];
+  }
+  return undefined;
+}
+
 function loadEnvironmentProxyValue(): string | undefined {
+  let httpProxy;
+  let httpsProxy;
+  let allProxy;
+
   if (!process) {
     return undefined;
   }
 
-  if (process.env[Constants.HTTPS_PROXY]) {
-    return process.env[Constants.HTTPS_PROXY];
-  } else if (process.env[Constants.HTTPS_PROXY.toLowerCase()]) {
-    return process.env[Constants.HTTPS_PROXY.toLowerCase()];
-  } else if (process.env[Constants.HTTP_PROXY]) {
-    return process.env[Constants.HTTP_PROXY];
-  } else if (process.env[Constants.HTTP_PROXY.toLowerCase()]) {
-    return process.env[Constants.HTTP_PROXY.toLowerCase()];
+  httpsProxy = environmentProxyValue(Constants.HTTPS_PROXY)!;
+  httpProxy = environmentProxyValue(Constants.HTTP_PROXY)!;
+
+  if (!httpProxy || !httpsProxy) {
+    allProxy = environmentProxyValue(Constants.ALL_PROXY)!;
+
+    if (!httpProxy) {
+      httpProxy = allProxy;
+    } else {
+      httpsProxy = allProxy;
+    }
+  }
+
+  if (!httpProxy && !httpsProxy) {
+    return undefined;
   }
 
-  return undefined;
+  return httpsProxy || httpProxy;
+}
+
+export function isBypassed(uri: string) {
+  let bypass = loadNoProxy();
```

Comment:
It feels that we should cache the result so the bypass list is not loaded every time `isByPassed()` is called.